src/main/java/com/conveyal/gtfs/validator/PostLoadValidator.java

```diff
@@ -0,0 +1,135 @@
+package com.conveyal.gtfs.validator;
+
+import com.conveyal.gtfs.GTFSFeed;
+import com.conveyal.gtfs.error.EmptyFieldError;
+import com.conveyal.gtfs.error.ForbiddenFieldError;
+import com.conveyal.gtfs.error.GeneralError;
+import com.conveyal.gtfs.error.RangeError;
+import com.conveyal.gtfs.error.ReferentialIntegrityError;
+import com.conveyal.gtfs.model.Stop;
+
+import java.util.List;
+
+/**
+ * Currently we perform a lot of validation while we're loading the rows out of the input GTFS feed.
+ * This can only catch certain categories of problems. Other problems must be found after tables are fully loaded.
+ * These include self-referential tables like stops (which reference other stops as parent_stations).
+ * This could also be expressed as a postLoadValidation method on Entity.Loader.
+ *
+ * In the original RDBMS-enabled gtfs-lib we had a lot of validation classes that would perform checks after loading.
+ * This included more complex semantic checks of the kind we do in r5 while building networks. We might want to
+ * re-import those gtfs-lib validators and adapt them to operate on MapDB only for the purposes of r5.
+ * However, validating a feed takes a lot of sorting and grouping of stop_times that will need to be repeated when we
+ * build a network. It's debatable whether we should make the user wait twice for this, as it's one of the slower steps.
+ */
+public class PostLoadValidator {
+
+    private GTFSFeed feed;
+
+    public PostLoadValidator (GTFSFeed feed) {
+        this.feed = feed;
+    }
+
+    public void validate () {
+        validateCalendarServices();
+        validateStopStationConstraints();
+    }
+
+    /**
+     * calendars and calendar_dates are the only conditionally required tables: at least one of the two must be present.
+     * The underlying requirement is that at least some service is defined.
+     */
+    private void validateCalendarServices () {
+        if (feed.services.isEmpty()) {
+            feed.errors.add(new GeneralError("calendar.txt", 0, "service_id",
+                    "Feed does not define any services in calendar or calendar_dates."));
+        }
+    }
+
+    /** Validate location_type and parent_station constraints as well as referential integrity. */
+    private void validateStopStationConstraints () {
+        for (Stop stop : feed.stops.values()) {
+            for (Rule rule : rules) {
+                if (rule.check(stop, feed)) break;
+            }
+        }
+    }
+
+    private static final String FILE = "stops.txt";
+    private static final String FIELD = "parent_station";
+
+    /** GTFS location_type codes. */
+    private enum LocType {
+        STOP(0),
+        STATION(1),
+        ENTRANCE(2),
+        GENERIC(3),
+        BOARDING(4);
+        public final int code;
+        LocType (int code) {
+            this.code = code;
+        }
+    }
+
+    private enum Requirement {
+        OPTIONAL, REQUIRED, FORBIDDEN
+    }
+
+    private static class Rule {
```

Comment:
At first I thought `Rule` was much more general purpose than the implementation here. Would `LocationTypeRule` or `StopStationRule` help avoid future confusion?